### PR TITLE
Fix proposal height set

### PR DESCRIPTION
**Pull request by kamilsa**
Signed-off-by: kamilsa <kamilsa16@gmail.com>

### Description of the Change

With current code it is possible that if proposal is not published proposal_height_ is still updated. That may cause Iroha to stop accepting new proposals due to invalid value of proposal height

### Benefits

Fixed problem

### Possible Drawbacks 

None

### CODE REVIEW

File: irohad/ordering/impl/single_peer_ordering_service.cpp [hunk 1]
```diff
@@ -112,9 +112,10 @@ namespace iroha {
             // In case of restart it reloads state.
             if (persistent_state_->createOsPersistentState() |
                 [this](const auto &state) {
-                  return state->saveProposalHeight(proposal_height_);
+                  return state->saveProposalHeight(proposal_height_+1);
```
Comment (l4l) [inline irohad/ordering/impl/single_peer_ordering_service.cpp]: May it save twice `X` value and then make height equals `X + 2`?

Comment (l4l) [reply irohad/ordering/impl/single_peer_ordering_service.cpp]: and put spaces around `+` plz

Comment (l4l) [inline irohad/ordering/impl/single_peer_ordering_service.cpp]: and put spaces around `+` plz

File: irohad/ordering/impl/single_peer_ordering_service.cpp [hunk 2]
```diff
@@ -103,7 +103,7 @@ namespace iroha {
 
       auto tx_range = txs | boost::adaptors::indirected;
       auto proposal = factory_->createProposal(
-          proposal_height_++, iroha::time::now(), tx_range);
+          proposal_height_, iroha::time::now(), tx_range);
 
       proposal.match(
```
Comment (igor-egorov) [inline irohad/ordering/impl/single_peer_ordering_service.cpp]: test comment

File: irohad/ordering/impl/single_peer_ordering_service.cpp [hunk 3]
```diff
@@ -103,7 +103,7 @@ namespace iroha {
...
           [this](expected::Value<
```
Comment (igor-egorov) [inline irohad/ordering/impl/single_peer_ordering_service.cpp]: test 2

File: irohad/ordering/impl/single_peer_ordering_service.cpp (hunk 1 above)
Comment (igor-egorov) [reply irohad/ordering/impl/single_peer_ordering_service.cpp]: That is interesting, do you have an idea how we can prevent this if the case with X in DB and X + 2 in memory is possible?

Comment (luckychess) [inline irohad/ordering/impl/single_peer_ordering_service.cpp]: I suggest use ```++proposal_height_ ``` here and get rid of line 118 increment.

Comment (kamilsa) [reply irohad/ordering/impl/single_peer_ordering_service.cpp]: Not sure if that will work. If expression in that line will fail, then we will not save proposal height, and increase it. This should not happen

**Pull request merged**